adr/2022_02_11_declaring_services_in_container.md

```diff
@@ -0,0 +1,65 @@
+# Declaring services in container
+
+* Status: accepted
+* Date: 2022-02-11
+
+## Context and Problem Statement
+
+Since Symfony 3.4, all Symfony services are declared private by default. And this is the recommended Symfony approach. On the 
+other hand, first stable Sylius release was published before that and therefore at the moment of publication of this recommendation
+Sylius had thousands (or at least hundreds) public services. The question is how should we handle services declared after the 
+3.4 release.
+
+## Decision Drivers
+
+Declared services should be:
+* services should be easily usable in ResourceControllers (where we are allowing for fetching services directly from container)
+* services should be easily usable in state machine callback (where we are allowing for fetching services directly from container)
+* it would be better to have one, coherent rule for all services
+* services should be easily accessible in PHPUnit
+
+## Considered Options
+
+You are right, this PR is against Symfony Best practices. However, our internal rule is to make all services public, 
+due to Sylius Resource Controller and testing with PHPUnit for example. In all these scenarios we are fetching services 
+directly from the container. In the resource controller, the services are fetched to perform regular logic. In the case 
+of PHPUnit, I just had a failing test at #12968. IIRC, Behat also requires us to have public services, but I'm not 100% sure about it.
```

Comment:
`I just had a failing test` - I believe ADRs should be impersonal

Reply:
`but I'm not 100% sure about it.` let's check it to be sure 😄  or if we're still not sure - let's not mention it here